app/models/script.rb

```diff
@@ -0,0 +1,24 @@
+class Script < ApplicationRecord
+
+  has_many :events
+
+  before_validation :add_or_update_checksum
+
+  validates_presence_of :text
+  validate :unique_text
+
+  private
+
+  def add_or_update_checksum
+    return nil unless self.text # avoid triggering error on Script.new.valid?
+    checksum = Digest::SHA1.hexdigest(self.text)
+    if self.checksum.nil? || self.checksum != checksum
+      self.checksum = checksum
+    end
+  end
```

Comment:
I'd argue that this seems a bit excessive. Hashing the contents of the text to avoid duplication wouldn't really avoid a lot of duplication, since omission of a single character would change the whole hash. There are other ways of doing this sort of text comparison, but ultimately I'd question whether we want this as a feature? @AaronSofaer can you weigh in on what the utility of this de-duplication is?